Initium-Odp/src/com/universeprojects/miniup/server/scripting/wrappers/Character.java

```diff
@@ -462,4 +460,72 @@ private ODPKnowledgeService getKnowledgeService() {
 		
 		return knowledgeService;
 	}
+	
+	/*################# QUESTS ###################*/	
+	/**
+	 * Grants a quest by name of the Def, and then returns it.
+	 * @param name
+	 * @return
+	 */
+	public Quest grantQuest(String name) {
+		if(true == hasQuest(name)) return getQuestByName(name);
+		
+		CachedEntity questDefEntity = db.getFilteredList("QuestDef", 1, "name", FilterOperator.EQUAL, name).get(0);
+		
+		return grantQuestInternal(questDefEntity);
+	}
+	
+	/**
+	 * Grants a quest based on the Def ID, then returns it.
+	 * @param defId
+	 * @return
+	 */
+	public Quest grantQuest(Long defId) {
+		return grantQuestInternal(db.getEntity(KeyFactory.createKey("QuestDef", defId)));
+	}
+	
+	private Quest grantQuestInternal(CachedEntity qde) {
+		if(qde == null) return null;
+		
+		QuestEntity raw = getQuestService().createQuestInstance(qde.getKey());
+				
+		return new Quest(raw, db);
+	}
+	
+	public boolean hasQuest(String name) {
+		return getQuestService().getQuestByName(name) != null;
+	}
+	
+	public Quest getQuestByName(String name) {
+		QuestEntity rawQuest = getQuestService().getQuestByName(name);
+		if(rawQuest == null) return null;
+		return new Quest(rawQuest, db);
+	}
+	
+	public Quest[] getAllQuests() {
+		List<QuestEntity> allQuests = getQuestService().getAllQuests();
+		List<Quest> toReturn = new ArrayList<>();
+		
+		for(QuestEntity quest : allQuests)
+			toReturn.add(new Quest(quest, db));
+		
+		return toReturn.toArray(new Quest[toReturn.size()]);
+	}
+	
+	public Quest[] getActiveQuests() {
+		List<QuestEntity> activeQuests = getQuestService().getActiveQuests();
+		List<Quest> toReturn = new ArrayList<>();
+		
+		for(QuestEntity quest : activeQuests)
+			toReturn.add(new Quest(quest, db));
+		
+		return toReturn.toArray(new Quest[toReturn.size()]);
+	}
+	
+	private QuestService getQuestService() {
+		if(questService == null)
+			questService = db.getQuestService(null);
+		
+		return questService;
+	}

```

Comment:
If possible, try to separate DA from the actual POCO objects. What if we add Quest support to items or locations? Having a separate location for the DA would require changing in only 1 place.